libc/misc/stat.c

```diff
@@ -500,9 +501,12 @@ TEST(stat_mode, fifo_type)
 	TEST_ASSERT_EQUAL_INT(0, lstat(tempPath, &buffer));
 	TEST_ASSERT_TRUE((buffer.st_mode & S_IFMT) == S_IFIFO);
 
+	fifo_fd = open(tempPath, O_NONBLOCK);
```

Comment:
Is it done 2 times?

Reply:
One for fifo created with 777 permissions, and second time for fifo with 000 permissions

Reply:
Right, it wasn't visible by dafault while reviewing, LGMT